corehq/apps/importer/util.py

```diff
@@ -404,12 +407,16 @@ def populate_updated_fields(config, columns, row, datemode):
 
         if update_value is not None:
             if field_map[key]['type_field'] == 'date':
-                update_value = parse_excel_date(update_value, datemode)
-            elif field_map[key]['type_field'] == 'integer':
                 try:
-                    update_value = str(int(update_value))
-                except ValueError:
-                    update_value = ''
+                    update_value = parse_excel_date(update_value, datemode)
+                except Exception:
+                    raise InvalidDateException(key)
+            elif field_map[key]['type_field'] == 'integer':
+                if str(update_value).strip() != '':
```

Comment:
I think what this does is what you are suggesting (it is a bit confusing because of the redefinition of `update_value`)
if the `update_value` is blank then don't try and coerce to an integer, `update_value` stays blank, and propagates to wherever the update happens; otherwise, if it is not blank, try and coerce it to an integer, and catch an error if that occurs. 

I'll tried writing a test for this for about 2 hours... but then gave up as I was getting weird errors with the way the tests handle updates. Probably something to look into, but out of scope for this I think.

I did test your usecase manually though, and it does what you suggest it should.